### PR TITLE
Refactor `spec_helper.rb` to remove use of `$user_1` and `$user_2`

### LINKED ISSUE #6094: Extract $user_1 and $user_2 creation
[Spec helper always create $user_1 and $user_2](https://github.com/CartoDB/cartodb/blob/master/spec/spec_helper.rb#L87), which is slow and not always needed. Tests should better create their own test data.

### PR DESCRIPTION
**Pull request by gfiorav**
Fixes #6094

### CODE REVIEW

**Comment by gfiorav** [comment]
It's not without great pain in my heart, I assign you this CR, @juanignaciosl 

4 years of tests refactor (shudder)

File: spec/factories/users_helper.rb [hunk 1]
```diff
@@ -9,8 +9,8 @@
 
   before(:all) do
     # TODO: Remove this and either all use the global instances or create a true general context with sample users
```
Comment (juanignaciosl) [inline spec/factories/users_helper.rb]: You can now delete this comment :clap:

**Comment by juanignaciosl** [comment]
:+1: :clap: You've earned the [chinese patience badge](http://www.career-success-for-newbies.com/chinese-symbol-for-patience.html).

File: spec/factories/users_helper.rb (hunk 1 above)
Comment (gfiorav) [inline spec/factories/users_helper.rb]: :__________________________)

**Comment by gfiorav** [comment]
@juanignaciosl I'm honoured :cry:

**Pull request merged**